src/components/sidebar/LineItem.tsx

```diff
@@ -1,39 +1,122 @@
 import { inject, observer } from 'mobx-react';
 import * as React from 'react';
 import classNames from 'classnames';
+import { FaAngleDown, FaAngleUp } from 'react-icons/fa';
+import { NotificationStore } from '../../stores/notificationStore';
 import { RouteStore } from '../../stores/routeStore';
 import lineHelper from '../../util/lineHelper';
 import { LineStore } from '../../stores/lineStore';
-import { ILine, IRoute } from '../../models';
+import { ILine, IRoute, ILineRoute } from '../../models';
 import RouteService from '../../services/routeService';
 import TransitTypeColorHelper from '../../util/transitTypeColorHelper';
 import Moment from 'react-moment';
 import * as s from './lineItem.scss';
+import NotificationType from '../../enums/notificationType';
+import LineItemSubMenu from './LineItemSubMenu';
 
 interface ILineItemState {
-    type: string;
+    openRouteIds: string[];
 }
 
 interface ILineItemProps {
+    notificationStore?: NotificationStore;
     lineStore?: LineStore;
     routeStore?: RouteStore;
     line: ILine;
 }
 
+@inject('notificationStore')
 @inject('lineStore')
 @inject('routeStore')
 @observer
 class LineItem extends React.Component<ILineItemProps, ILineItemState> {
-    private selectRoute(routeId: string) {
+    constructor(props: ILineItemProps) {
+        super(props);
+        this.state = {
+            openRouteIds: [],
+        };
+    }
+
+    private selectRoute(routeId: string, e: any) {
         RouteService.getRoute(this.props.line.lineId, routeId)
             .then((res: IRoute) => {
                 this.props.routeStore!.addToRoutes(res);
                 this.props.lineStore!.setSearchInput('');
             })
             .catch((err: any) => {
+                this.props.notificationStore!.addNotification({
+                    message: 'Reitin haussa tapahtui virhe.',
+                    type: NotificationType.ERROR,
+                });
             });
     }
 
+    private isRouteOpen(routeId: string) {
+        return this.state.openRouteIds.some(id => id === routeId);
+    }
+
+    private openRouteMenu(routeId: string) {
+        this.setState({
+            openRouteIds: this.state.openRouteIds.concat(routeId),
+        });
+    }
+
+    private closeRouteMenu(routeId: string) {
+        this.setState({
+            openRouteIds: this.state.openRouteIds.filter(id => id !== routeId),
+        });
+    }
+
+    private toggleRouteMenu(routeId: string, e: any) {
+        e.stopPropagation();
+        if (this.isRouteOpen(routeId)) {
+            this.closeRouteMenu(routeId);
+        } else {
+            this.openRouteMenu(routeId);
+        }
+    }
+
+    private getRoute(route: ILineRoute) {
```

Comment:
How about renderRoute? Now getRoute() gives the idea of fetching route from GraphQL.

For methods that return JSX, i would put render prefix